### PR TITLE
fix(sql): add check for GroupBy function at argument position.

### PR DESCRIPTION
**Pull request by eugenels**
This PR adds early check for "GroupBy" function at the argument position.
It throws exception instead of segfault.

Fixes: https://github.com/questdb/questdb/issues/1806

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|6|6|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|6|6|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|6|6|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|6|6|100.00%|

File: core/src/main/java/io/questdb/griffin/FunctionParser.java [hunk 1]
```diff
@@ -252,8 +253,15 @@ public void visit(ExpressionNode node) throws SqlException {
             mutableArgPositions.clear();
             mutableArgPositions.setPos(argCount);
             for (int n = 0; n < argCount; n++) {
-                mutableArgs.setQuick(n, functionStack.poll());
-                mutableArgPositions.setQuick(n, positionStack.pop());
+                final Function arg = functionStack.poll();
+                final int pos = positionStack.pop();
+
+                if (arg instanceof GroupByFunction) {
+                    throw SqlException.position(pos).put("Aggregate function cannot be passed as an argument");
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/FunctionParser.java:260]: `Function` is generally `closeable`. Some implementations allocate memory. When exception is thrown we need to ensure that functions are released.

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/questdb/griffin/FunctionParser.java (hunk 1 above)
Comment (eugenels) [reply core/src/main/java/io/questdb/griffin/FunctionParser.java:260]: It's not very obvious what to clear at this point: `mutableArgs`, top of the `functionStack` or all items inside of it?
Is it safe to close Function several times?
Coz Function reference can be at different places.

Also, we have several throws places later in createFuntion:
https://github.com/questdb/questdb/blob/c4275e41560494cecdc350319d1aa8fde46bb16e/core/src/main/java/io/questdb/griffin/FunctionParser.java#L460
https://github.com/questdb/questdb/blob/c4275e41560494cecdc350319d1aa8fde46bb16e/core/src/main/java/io/questdb/griffin/FunctionParser.java#L646
https://github.com/questdb/questdb/blob/c4275e41560494cecdc350319d1aa8fde46bb16e/core/src/main/java/io/questdb/griffin/FunctionParser.java#L639
https://github.com/questdb/questdb/blob/c4275e41560494cecdc350319d1aa8fde46bb16e/core/src/main/java/io/questdb/griffin/FunctionParser.java#L320

Comment (bluestreak01) [reply core/src/main/java/io/questdb/griffin/FunctionParser.java:260]: That is very cool way to present a comment, thank you 🎉 

As long as you're confident we're not going to leak memory out of allocating function, I'm happy to trust your judgement

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 25 / 25 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|9|9|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 30 / 30 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|19|19|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|11|11|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 30 / 30 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|19|19|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|11|11|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 30 / 30 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|19|19|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|11|11|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 30 / 30 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|19|19|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|11|11|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 30 / 30 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|19|19|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|11|11|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 30 / 30 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|19|19|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|11|11|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**